tools/shared/src/main/scala/org/scalajs/core/tools/linker/checker/IRChecker.scala

```diff
@@ -312,41 +316,47 @@ private final class IRChecker(unit: LinkingUnit, logger: Logger) {
           "but must be NoType")
     }
 
-    val bodyStats = body match {
-      case Block(stats) => stats
-      case _            => body :: Nil
-    }
+    body match {
+      case None =>
```

Comment:
Not really. Short of side-effectness... Style guide actually says no use of fold for side effects... Although we do...